src/plone/api/content.py

```diff
@@ -473,3 +474,45 @@ def get_uuid(obj=None):
     :Example: :ref:`content_get_uuid_example`
     """
     return IUUID(obj)
+
+
+def find(context=None, depth=None, **kwargs):
+    """Find content
+
+    :param context: Context within we want to search
+    :type obj: Content object
+    :param depth: How far in the content tree we want to search from context
+    :type obj: Content object
+    :returns: Catalog brains
+    :rtype: List
+    :Example: :ref:`content_find_example`
+    """
+    query = {}
+    query.update(**kwargs)
+
+    # Passing a context or depth overides the exising path query
+    if context or depth:
+        query['path'] = {}
+
+    # Limit search depth
+    if depth is not None:
+        # If we don't have a context, we'll assume the portal root.
+        if context is None:
```

Comment:
Right, from reading the tests, it appears you have done this because you'd like to be able to pass in object_provides and get all instances of that class across the whole site. We should in that case probably consider making either context *or* object_provides required.